assets/js/base/hooks/shipping/use-shipping-address.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useReducer } from '@wordpress/element';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { useDebounce } from 'use-debounce';
+import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+
+/**
+ * Internal dependencies
+ */
+import { useStoreCart } from '../cart/use-store-cart';
+import { pluckAddress } from '../../utils';
+
+const reducer = ( state, address ) =>
+	isShallowEqual( pluckAddress( address ), pluckAddress( state ) )
+		? state
+		: { ...state, ...address };
+
+export const useShippingAddress = () => {
+	const { shippingAddress: initialAddress } = useStoreCart();
+	const [ shippingAddress, setShippingAddress ] = useReducer(
```

Comment:
what I do love about this new approach is that is somehow solved an issue with the previous approach, shipping fields in checkout didn’t populate if Cart endpoint returned an address, now it populates it 🙌